scripts/run-eslint.ts

```diff
@@ -274,21 +368,31 @@ function saveWarningThresholds(newWarningCounts: WarningCounts): void {
  * Given a list of results from an the ESLint run, counts the number of warnings
  * produced per file and rule.
  *
- * @param results - The ESLint results.
+ * @param args - The arguments.
+ * @param args.results - The results from running `eslint`.
+ * param args.countFileEvenIfNoWarnings - Includes a file in the returned
+ * object even if there are no recorded warnings for it.
  * @returns A two-level object mapping path to files in which warnings appear to
  * the IDs of rules for those warnings, then from rule IDs to the number of
  * warnings for the rule.
  */
-function getWarningCounts(results: ESLint.LintResult[]): WarningCounts {
+function getWarningCounts({
+  results,
+}: {
+  results: ESLint.LintResult[];
+}): WarningCounts {
   const unsortedWarningCounts = results.reduce(
     (workingWarningCounts, result) => {
       const { filePath } = result;
       const relativeFilePath = path.relative(PROJECT_DIRECTORY, filePath);
+      if (!workingWarningCounts[relativeFilePath]) {
```

Comment:
This change is necessary now that we are always merging what this function returns with what already exists in the warning thresholds file (see above).

If all of the lint violations that are warnings are fixed within a file, we want to include that file path in the object that this function returns because otherwise the counts won't get reset for that file in the warning thresholds file and our script will still think that there are warnings.